### PR TITLE
ZBUG-382: zimbraMailSieveScript attribute needs to be limited by the number of conditions and rules

### PR DESCRIPTION
**Pull request by preshitaagrawal**
Ported changes by cherrypicking.

### CODE REVIEW

File: WebRoot/js/zimbraMail/prefs/model/ZmFilterRules.js [hunk 1]
```diff
@@ -400,15 +400,14 @@ function(ex) {
...
-			case ZmCsfeException.MAIL_NO_SUCH_TAG:
-				msg = ZmMsg.errorNoSuchTag;
-				break;
-			default:
-				msg = [ZmMsg.filterError, " ", AjxStringUtil.htmlEncode(ex.msg)].join("");
+		if (ex.code === ZmCsfeException.MAIL_NO_SUCH_FOLDER) {
+			msg = ZmMsg.errorNoSuchFolder;
+		} else if (ex.code === ZmCsfeException.MAIL_NO_SUCH_TAG) {
+			msg = ZmMsg.errorNoSuchTag;
+		} else if (ex.code === ZmCsfeException.SVC_INVALID_REQUEST && ex.msg.match(ZmMsg.filterServiceErrorMatchString)) {
```
Comment (silentsakky) [inline WebRoot/js/zimbraMail/prefs/model/ZmFilterRules.js:407]: Can't we create a new type of exception for this scenario like ZmCsfeException.SIEVE_EXCEPTION? doing a regex match on a server exception msg is usually not preferred

**Comment by k-kato** [comment]
please update description and commit comments as discussed in a different channel.

File: WebRoot/js/zimbraMail/prefs/model/ZmFilterRules.js [hunk 2]
```diff
@@ -393,22 +393,21 @@ function(index, notify, callback, result) {
...
-		switch(ex.code) {
-			case ZmCsfeException.MAIL_NO_SUCH_FOLDER:
-				msg = ZmMsg.errorNoSuchFolder;
-				break;
-			case ZmCsfeException.MAIL_NO_SUCH_TAG:
-				msg = ZmMsg.errorNoSuchTag;
-				break;
-			default:
-				msg = [ZmMsg.filterError, " ", AjxStringUtil.htmlEncode(ex.msg)].join("");
+		if (ex.code === ZmCsfeException.MAIL_NO_SUCH_FOLDER) {
```
Comment (silentsakky) [inline WebRoot/js/zimbraMail/prefs/model/ZmFilterRules.js:403]: not sure if we want to change switch statement to if/else now

Comment (preshitaagrawal) [reply WebRoot/js/zimbraMail/prefs/model/ZmFilterRules.js]: corrected..

**Review by miteshsavani810 — APPROVED** [review]

File: WebRoot/messages/ZmMsg.properties [hunk 3]
```diff
@@ -1463,6 +1463,7 @@ filterErrorCondition = The condition "{0}" is missing one or more values.
 filterErrorIllegalCharacter = Filter values may not contain double-quote (") or backslash (\\)
 filterErrorNameExists = There is already a filter by that name.
 filterErrorNoName = The new filter needs a name.
+filterErrorSieveScriptMaxSize = Filter size limit exceeded. Please examine the filter contents and reduce the number of settings.
```
Comment (k-kato) [inline WebRoot/messages/ZmMsg.properties:1466]: Will we translate it on the ticket or on a separate ticket?

**Review by abhishekdeshmukh2209 — APPROVED** [review]

File: WebRoot/messages/ZmMsg.properties (hunk 3 above)
Comment (k-kato) [reply WebRoot/messages/ZmMsg.properties:1466]: okay. Please create a ticket and write the ticket number here so that we will not miss it.

**Review by k-kato — APPROVED** [review]

**Review by damini-vashishtha — APPROVED** [review]

**Pull request merged**